File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/mining/smartcontract/SCExecutedTraceEvent.java
package org.palladiosimulator.blockchainsystems.core.mining.smartcontract;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction;
import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent;

public record SCExecutedTraceEvent(
        long occurrenceTime,
        Transaction transaction
) implements TraceEvent {

    public static final String EVENT_TYPE = "SCExecutedTraceEvent";

    @Override
    public long getOccurrenceTime() {
        return this.occurrenceTime();
    }

    @Override
    public String getEventType() {
        return EVENT_TYPE;
    }

    @Override
    public void formatDetails(StringBuilder stringBuilder) {
        stringBuilder.append("{ txId=");
        stringBuilder.append(transaction.getTxId());
        stringBuilder.append(" }");
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/mining/smartcontract/SCExecutedTraceEvent.kt
package org.palladiosimulator.blockchainsystems.core.mining.smartcontract

import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction
import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent

/**
 * @author Davis Riedel
 */
class SCExecutedTraceEvent(
  private val occurrenceTime: Long,
  private val transaction: Transaction
) : TraceEvent {
  companion object {
    const val EVENT_TYPE = "SCExecutedTraceEvent"
  }

  override fun getOccurrenceTime(): Long = occurrenceTime
  override fun getEventType(): String = EVENT_TYPE

  override fun formatDetails(stringBuilder: StringBuilder) {
    stringBuilder.append("{ txId=")
    stringBuilder.append(transaction.txId)
    stringBuilder.append(" }")
  }
}
